packages/ember-glimmer/lib/environment.js

```diff
@@ -68,7 +88,8 @@ export default class Environment extends GlimmerEnvironment {
         let definition = this.getComponentDefinition(path);
 
         if (definition) {
-          return new CurlyComponentSyntax({ args, definition, templates });
+          wrapClassAttribute(args);
```

Comment:
It also only wraps cases were the binding is used directly. For example `{{foo-bar class=(if fooBar "lol")}}` will not get wrapped. We are changing the syntax at compile time.